lib/core/include/irods/irods_server_properties.hpp

```diff
@@ -136,18 +148,37 @@ namespace irods
 
         void remove( const std::string& _key );
 
+        /// \brief Get a reference to the underlying json object. Writing
+        /// in particular should be done with caution as there is no guarantee
+        /// that this will be safe.
```

Comment:
Add a line stating use of this function bypasses safety guarantees when called in an asynchronous context.